src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -875,6 +875,31 @@ class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
     })
   }
 
+  override def listResourceChildren(resource: FullyQualifiedResourceId, samRequestContext: SamRequestContext): IO[Set[FullyQualifiedResourceId]] = {
+    val r = ResourceTable.syntax("r")
+    val rt = ResourceTypeTable.syntax("rt")
+    val cr = ResourceTable.syntax("cr")
+
+    val query =
+      samsql"""
+         select ${cr.result.name}, ${rt.result.name}
+         from ${ResourceTable as r}
+         join ${ResourceTable as cr} on ${cr.resourceParentId} = ${r.id}
+         join ${ResourceTypeTable as rt} on ${rt.id} = ${cr.resourceTypeId}
+         where ${r.name} = ${resource.resourceId}
+         and ${rt.name} = ${resource.resourceTypeName}"""
```

Comment:
We only want to list child resources with the same Resource Type as the parent?  This surprised me when I read it.  It may be what we want, but it's not what I was expecting.  

If this is what we want, we may want to add a comment here and possibly update the swagger description of the api to explain that it will return only children of the same Resource Type.  

Also, if this is what we want, we should right a test where a resource has children with varied Resource Types.

Reply:
I don't think that's what we want, that would be an unintentional feature (bug) in my query. I'll add a test to make sure we aren't doing that and adjust the query